src/mono/netcore/sample/Android/AndroidSampleApp.csproj

```diff
@@ -48,19 +48,39 @@
     <Message Importance="High" Text="Apk:       $(ApkBundlePath)"/>
     <Message Importance="High" Text="PackageId: $(ApkPackageId)"/>
 
-    <Exec Condition="'$(DeployAndRun)' == 'true'" Command="$(AdbTool) kill-server"/>
-    <Exec Condition="'$(DeployAndRun)' == 'true'" Command="$(AdbTool) start-server"/>
-    <Exec Condition="'$(DeployAndRun)' == 'true'" Command="$(AdbTool) logcat -c" ContinueOnError="WarnAndContinue" />
-    <Message Condition="'$(DeployAndRun)' == 'true'" Importance="High" Text="Uninstalling apk (ignore errors if any):"/>
-    <Exec Condition="'$(DeployAndRun)' == 'true'" Command="$(AdbTool) uninstall net.dot.HelloAndroid" ContinueOnError="WarnAndContinue" />
-    <Exec Condition="'$(DeployAndRun)' == 'true'" Command="$(AdbTool) install $(ApkDir)/bin/HelloAndroid.apk" />
-    <Exec Condition="'$(DeployAndRun)' == 'true' and '$(RunActivity)' != 'true'"
+    <Exec Condition="'$(DeployAndRun)' == 'true' and '$(ArchiveTests)' != 'true'" Command="$(AdbTool) kill-server"/>
```

Comment:
Is this ArchiveTests condition basically used to distinguish between weather the sample is being run as part of the test suite or as an independent program? I am not sure I understand the name.

Reply:
I feel that it is an indicator for running the sample on CI or locally. But I might be wrong.

Reply:
Yes, it tells us if we're running as a functional test or independently.

Reply:
@steveisok Is there some API forcing the use of the name "ArchiveTests"? It doesn't seem like it has much to do with "Archiving".

Reply:
The name pre-dates me.  But it's used in our msbuild testing infra, so yes to your question.

Reply:
Well, I'd consider a comment since its a little unclear when than ArchiveTests is true.

Reply:
> Is this ArchiveTests condition basically used to distinguish between whether the sample is being run as part of the test suite or as an independent program?  

Yes, the `ArchiveTests` condition is just a good means to incorporate the sample into the library test build. It also allows to preserve the existing setup to run the sample independently via make file.